packages/web-components/fast-components/src/color/recipes/contrast-set-recipe.ts

```diff
@@ -0,0 +1,45 @@
+import { Palette } from "../palette";
+import { InteractiveSwatchSet } from "../recipe";
+import { Swatch } from "../swatch";
+import { directionByIsDark } from "../utilities/direction-by-is-dark";
+
+/**
+ * @internal
+ */
+export function contrastSetRecipe(
+    palette: Palette,
+    reference: Swatch,
+    baseContrast: number,
+    restDelta: number,
+    hoverDelta: number,
+    activeDelta: number,
+    focusDelta: number
+): InteractiveSwatchSet {
+    const direction = directionByIsDark(reference);
+    const accessibleIndex = palette.closestIndexOf(
+        palette.colorContrast(reference, baseContrast)
+    );
+    const accessibleIndex2 =
+        accessibleIndex + direction * Math.abs(restDelta - hoverDelta);
+    const indexOneIsRest =
+        direction === 1
+            ? restDelta < hoverDelta
+            : direction * restDelta > direction * hoverDelta;
+    let restIndex: number;
+    let hoverIndex: number;
+
+    if (indexOneIsRest) {
+        restIndex = accessibleIndex;
+        hoverIndex = accessibleIndex2;
+    } else {
+        restIndex = accessibleIndex2;
+        hoverIndex = accessibleIndex;
+    }
```

Comment:
It's a little hard to see what is going on, but if I'm reading it correctly, if direction === 1 rest gets the lower number and hover gets the higher number. Otherwise if direction === -1 rest gets the higher number and hover gets the lower number? If that is the case, you could use Math.min and Math.max based on direction to assign these.

```suggestion
    const restIndex: number = Math[direction ? "min" : "max"](accessibleIndex1, accessibleIndex2);
    const hoverIndex: number = Math[direction ? "max" : "min"](accessibleIndex1, accessibleIndex2);
```